src/sentry/integrations/slack/notifications.py

```diff
@@ -32,26 +32,30 @@ def get_context(
 def get_channel_and_integration_by_user(
     user: User, organization: Organization
 ) -> Tuple[Optional[str], Optional[Integration]]:
-    try:
-        identity = Identity.objects.get(
-            idp__type=EXTERNAL_PROVIDERS[ExternalProviders.SLACK],
-            user=user.id,
-        )
-    except Identity.DoesNotExist:
+
+    identities = Identity.objects.filter(
+        idp__type=EXTERNAL_PROVIDERS[ExternalProviders.SLACK],
+        user=user.id,
+    )
+    if not identities:
         # The user may not have linked their identity so just move on
         # since there are likely other users or teams in the list of
         # recipients.
         return None, None
 
     try:
         integration = Integration.objects.get(
-            provider=identity.idp.type,
+            provider=EXTERNAL_PROVIDERS[ExternalProviders.SLACK],
             organizations=organization,
+            external_id__in=[identity.idp.external_id for identity in identities],
```

Comment:
This seems ok, I don't know whether we allow an org to have multiple slack integrations associated with it? That's the only thing I can think of that'd go wrong here.

Actually, now that I think about it, do we even need the external_id filtering here? This should always return a single row if we guarantee there's one slack integration per org.